tests/functional/conftest.py

```diff
@@ -346,17 +348,25 @@ def _run_executable(self, prog, args, run_from_path, runtime):
                 prog_cwd = prog_cwd.encode('mbcs')
 
         args = [prog_name] + args
-        # Run executable. stderr is redirected to stdout.
-        print('RUNNING: ', safe_repr(exe_path), ", args: ", safe_repr(args))
-
         # Using sys.stdout/sys.stderr for subprocess fixes printing messages in
         # Windows command prompt. Py.test is then able to collect stdout/sterr
         # messages and display them if a test fails.
+        for _ in range(_MAX_RETRIES):
+            retcode = self.__run_executable(args, exe_path, prog_env,
+                                            prog_cwd, runtime)
+            if retcode != -1:
+                break
+
+        return retcode
 
+    def __run_executable(self, args, exe_path, prog_env, prog_cwd, runtime):
         process = psutil.Popen(args, executable=exe_path,
                                stdout=subprocess.PIPE,
                                stderr=subprocess.PIPE,
                                env=prog_env, cwd=prog_cwd)
+        # Run executable. stderr is redirected to stdout.
+        print('[' + process.pid + '] RUNNING: ', safe_repr(exe_path),
```

Comment:
Is the PID really an useful information? Wouldn't this jsut result in output like this:
```
------- Starting build. -------
------- Build finshed, now running executable. -------
[321] RUNNING:  'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\pytest-of-appveyor\\pytest-0\\popen-gw2\\test_openpyxl_onefile_0\\dist\\test_openpyxl.exe' , args:  ['.\\test_openpyxl.exe']
[321] TIMED OUT:  'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\pytest-of-appveyor\\pytest-0\\popen-gw2\\test_openpyxl_onefile_0\\dist\\test_openpyxl.exe' , args:  ['.\\test_openpyxl.exe']
```

Reply:
It would output this:

```
[321] RUNNING:  'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\pytest-of-appveyor\\pytest-0\\popen-gw2\\test_openpyxl_onefile_0\\dist\\test_openpyxl.exe' , args:  ['.\\test_openpyxl.exe']
[321] TIMED OUT!
```

IMO it's cleaner because the args are printed once.

Reply:
ACK to printing the args only once. But I can not see any use in printing the process id?! What should it be good for?

Reply:
These messages are combined with the other messages on stdout. The PID distinguishes the messages coming from this code from the messages coming from the executable.